plugins/source_docs.go

```diff
@@ -14,20 +15,100 @@ import (
 //go:embed templates/*.go.tpl
 var templatesFS embed.FS
 
+type SourceDocsFormat string
+
+const (
+	SourceDocsFormatMarkdown = "markdown"
+	SourceDocsFormatJSON     = "json"
+)
+
+var SourceDocsFormats = []SourceDocsFormat{
+	SourceDocsFormatMarkdown,
+	SourceDocsFormatJSON,
+}
+
+func (s SourceDocsFormat) String() string {
+	return string(s)
+}
+
+func (s SourceDocsFormat) Validate() error {
```

Comment:
I think this can be just an enum (https://github.com/cloudquery/plugin-sdk/blob/main/serve/source.go#L67) which already include the validation.

Reply:
Good point, but I think the enum code is private to the `serve` package so I'll have to refactor a bit first.